runtime/compiler/microjit/x/amd64/AMD64Codegen.hpp

```diff
@@ -0,0 +1,303 @@
+            TR_J9ByteCode bc,
+            TR_J9ByteCodeIterator* bci
+        );
+
+        /**
+         * Generates a store instruction based on the type.
+         *
+         * @param buffer         code buffer
+         * @param method         method for introspection
+         * @param bc             Bytecode that generated the load instruction
+         * @param bci            Bytecode iterator from which the bytecode was retreived
+         * @return               size of generated code; 0 if method failed
+         */
+        buffer_size_t
+        generateStore(
+            char* buffer,
+            TR_ResolvedMethod* method,
+            TR_J9ByteCode bc,
+            TR_J9ByteCodeIterator* bci
+        );
+
+        /**
+         * Generates a store instruction based on the type.
+         *
+         * @param buffer         code buffer
+         * @param method         method for introspection
+         * @param bci            Bytecode iterator from which the bytecode was retreived
+         * @return               size of generated code; 0 if method failed
+         */
+        buffer_size_t
+        generatePutStatic(
+            char* buffer, 
+            TR_ResolvedMethod* method, 
+            TR_J9ByteCodeIterator* bci
+        );
+
+        /**
+         * Generates a store instruction based on the type.
+         *
+         * @param buffer         code buffer
+         * @param method         method for introspection
+         * @param bci            Bytecode iterator from which the bytecode was retreived
+         * @return               size of generated code; 0 if method failed
+         */
+        buffer_size_t
+        generateGetStatic(
+            char* buffer, 
+            TR_ResolvedMethod* method, 
+            TR_J9ByteCodeIterator* bci
+        );
+
+        /**
+         * Generates a return instruction based on the type.
+         * 
+         * @param buffer        code buffer
+         * @param dt            The data type to be returned by the TR-compiled method
+         * @return              size of generated code; 0 if method failed
+         */
+        buffer_size_t
+        generateReturn(
+            char* buffer, 
+            TR::DataType dt
+        );
+
+        LocalTable makeLocalTable(TR_J9ByteCodeIterator*, MJIT::LocalTableEntry*, U_16, int32_t);
+
+    public:
+        CodeGenerator() = delete;
+        CodeGenerator(J9MicroJITConfig*, J9VMThread*, TR::FilePointer*, TR_J9VMBase&, ParamTable*, TR::Compilation*, MJIT::CodeGenGC*);
+
+        inline void 
+        setPeakStackSize(int32_t newSize)
+        {
+            _stackPeakSize = newSize;
+        }
+
+        inline int32_t 
+        getPeakStackSize()
+        {
+            return _stackPeakSize;
+        }
+
+        buffer_size_t 
+        generatePrePrologue(
+            char*,
+            J9Method*,
+            char**,
+            char**,
+            TR_PersistentJittedBodyInfo**
+        );
```

Comment:
Parameter names would be helpful for all these methods to clarify the intent of each formal.